zgw/portaaltaak/src/test/kotlin/com/ritense/portaaltaak/PortaaltaakPluginTest.kt

```diff
@@ -0,0 +1,326 @@
+            processDocumentService,
+            zaakInstanceLinkService,
+            mock()
+        )
+        portaaltaakPlugin.notificatiesApiPluginConfiguration = mock()
+        portaaltaakPlugin.objectManagementConfigurationId = mock()
+    }
+
+    @Test
+    fun `should create taak object in objects api`() {
+        //todo incomplete
+        val formType = TaakFormType.ID
+        val formTypeId = "formTypeId"
+        val formTypeUrl = "formTypeUrl"
+        val sendData = emptyList<DataBindingConfig>()
+        val receiveData = emptyList<DataBindingConfig>()
+        val receiver = TaakReceiver.ZAAK_INITIATOR
+        val otherReceiver = OtherTaakReceiver.BSN
+        val kvk = null
+        val zakenApiPlugin = mock<ZakenApiPlugin>()
+        val objectenApiPlugin = mock<ObjectenApiPlugin>()
+
+        whenever(objectManagementService.getById(any())).thenReturn(getObjectManagement())
+        whenever(pluginService.createInstance(any<PluginConfigurationId>())).thenReturn(objectenApiPlugin)
+        whenever(pluginService.createInstance(any<Class<ZakenApiPlugin>>(), any())).thenReturn(zakenApiPlugin)
+        whenever(delegateTask.processInstanceId).thenReturn(UUID.randomUUID().toString())
+        whenever(zaakInstanceLinkService.getByDocumentId(any())).thenReturn(getZaakInstanceLink())
+        whenever(jsonSchemaDocumentId.id).thenReturn(UUID.randomUUID())
+        whenever(processDocumentService.getDocumentId(any(), any<DelegateTask>())).thenReturn(jsonSchemaDocumentId)
+        whenever(zakenApiPlugin.getZaakRollen(any(), any())).thenReturn(getRol(BetrokkeneType.NATUURLIJK_PERSOON))
+        whenever(delegateTask.name).thenReturn("delegateTaskName")
+        whenever(delegateTask.id).thenReturn("delegateTaskId")
+        whenever(jsonSchemaDocumentId.toString()).thenReturn("documentId")
+        whenever(valueResolverService.resolveValues(any(), any())).thenReturn(emptyMap())
+
+        portaaltaakPlugin.createPortaalTaak(
+            delegateTask,
+            formType,
+            formTypeId,
+            formTypeUrl,
+            sendData,
+            receiveData,
+            receiver,
+            otherReceiver,
+            kvk,
+            bsn
+        )
+        verify(pluginService, times(1)).createInstance(any<PluginConfigurationId>())
+        verify(pluginService, times(1)).createInstance(any<Class<ZakenApiPlugin>>(), any())
+        verify(objectManagementService, times(1)).getById(any())
+        verify(delegateTask, times(2)).processInstanceId
+        verify(delegateTask, times(1)).id
+        verify(delegateTask, times(1)).name
+        verify(zaakInstanceLinkService, times(1)).getByDocumentId(any())
+        verify(jsonSchemaDocumentId, times(1)).id
+        verify(processDocumentService, times(2)).getDocumentId(any(), any<DelegateTask>())
+        verify(zakenApiPlugin, times(1)).getZaakRollen(any(), any())
+        verify(valueResolverService, times(1)).resolveValues(any(), any())
```

Comment:
What are we testing here? For unit tests, you don't want to test if the code that is written matches exactly what you think it should be. Instead, try to write tests in such a way you can verify the output is what you expect (given certain scenarios).